pulpcore/cli/container/repository.py

```diff
@@ -119,3 +119,42 @@ def sync(
         href=repository_href,
         body=body,
     )
+
+
+@repository.command(name="tag")
+@name_option
+@href_option
+@click.option("--tag", help=_("Name to tag an image with"), required=True)
+@click.option("--digest", help=_("SHA256 digest of the Manifest file"), required=True)
+@pass_repository_context
+def add_tag(repository_ctx: PulpRepositoryContext, digest: str, tag: str,) -> None:
+    if len(tag) == 0:
+        raise click.ClickException("Please pass a non empty tag name.")
+
+    digest = digest.strip()
+    if not digest.startswith("sha256:"):
+        digest = f"sha256:{digest}"
+    if len(digest) != 71:  # len("sha256:" + 64
+        raise click.ClickException("Improper SHA256, please provide a valid 64 digit digest.")
```

Comment:
Tried doing some bare minimum checks and leaving the rest to the endpoint's serializer. If there is anything else needed please tell me.